### DISCUSSION TITLE
Error: Object reference not set to an instance of an object

### DISCUSSION THREAD

**Discussion by gr8bigguy** [TumblThreeApp/TumblThree#383] [Q&A (Question & Answer)]
I am a first time user and don't know if this error is one that appears because of "user error" and I just need to fix a setting or if it's more involved that requires steadier hands and more know-how.  There are pics on this blog to be downloaded. Help.

**Comment by thomas694** [comment]
Actually there are two errors with nearly the same text at the moment. We'll make the shown messages more clear and fix the first error.

The second isn't an error in TumblThree. You can try to load the blog in the browser while logged in. It doesn't load posts and at the bottom it says "Your search took too long". Technically their server is returning an error code 2. At least that's how it showed here.
Right now it seems to work again.

**Reply by gr8bigguy** [reply]
Not sure about how to load the blog in the browser while logged in.  Does that suggest, I wasn't logged in to newtumbl when I loaded the blog in newtumbl?  

Now I get another error: Could not load file in library: K:\NewTumbl\Index\blogname_files.newtumbl

**Comment by thomas694** [comment]
I meant opening the blog URL in a normal web browser. That way I saw that this error wasn't related to the app, but that their server had a temporary error.

**Reply by gr8bigguy** [reply]
Yeah, I opened all of the blogs in MS Edge first.  Then, I cut/paste the URL into the NewTumbl in the Enter URL box at the lower left corner of the app and then select Add Blog and Add to Queue.

**Reply by gr8bigguy** [reply]
Am I missing something?

**Comment by thomas694** [comment]
Please update and test with the newest version.

**Reply by gr8bigguy** [reply]
Says WS.reputation.1 (low reputation) "malware" is attached.  But also has a hard time uncompressing.  Will try again tomorrow.

**Reply by thomas694** [reply]
Oh! It's a false positive. Rest assured it is clean.
In short, already in 2.9.0 we added a process to clean up old files that are no longer needed by the new version.

**Comment by MSWallack** [comment]
I get a similar error when I go to Settings and try to authenticate my Tumblr login. The authentication window opens but then just stays blank and nothing happens. When I close the authentication window, I get the error.

**Reply by thomas694** [reply]
Do you use 2.9.0 or 2.9.1? Have you downloaded and installed the Webview2 runtime?

**Reply by MSWallack** [reply]
I had the same error with both 2.9.0 and 2.9.1. I have not installed the Webview2 runtime. Is that something that I need? I've been using TumblThree for ... a long time and never had this problem until it started in the last 10 days or so and, at least as far as I recall, I've never installed the Webview2 runtime. I also got an error last night (which I sadly forgot to write down; hey, it was late) that said something like TumblThree or Edge was unable to write to a particular file or folder.

**Reply by thomas694** [reply]
Yes, since 2.9.0 the Webview2 browser component is used, therefore the runtime needs to be installed, if not done before.
The user should get a reminder dialog on app start and when trying to open the Authenticate window. As you can open the dialog it means it found a runtime (which you could verify `Programs and Features` in `Control Panel`).
For the reasons why the dialog remains white and the error yesterday, please have a look in the log file (%LOCALAPPDATA%\TumblThree\TumblThree.log), whether you can see the error.

**Reply by MSWallack** [reply]
Apparently Webview2 was already installed so I did a repair installation. The error persisted in TumblThree. I closed the app and rebooted and ... fell asleep. I'll check on whether it's working after work.

**Reply by MSWallack** [reply]
After rebooting and restarting the application, I still can't authenticate. TumblThree pops up an error message that says:

> We couldn't create the data directory
> 
> Microsoft Edge can't read and write to its data directory:
> 
> C:\Program Files\TumblThree\TumblThree.exe.WebView2\EBWebView

That directory existed but was empty. I tried deleting it and then authenticating, but it still doesn't work.

**Comment by thomas694** [comment]
Ok, now I see the problem. We try to fix it in the next version.
You can add write permissions for the user group "Everyone" to the folder "C:\Program Files\TumblThree\TumblThree.exe.WebView2" and below. Or you can copy the installation folder to a non-protected location (e.g. C:\TumblThree) and use TumblThree from there until the next version comes.

**Reply by MSWallack** [reply]
Excellent. Thank you!

**Reply by gr8bigguy** [reply]
OK...cuz I've downloaded 2.9.1 with the Webview2 installed and I'm still getting the same initial error....and I'm logged in running ThumblThree with Administrator rights.  Still not downloading everything Thomas.  When do you anticipate either 2.9.2 or 3.0 coming out?

**Comment by thomas694** [comment]
Which error do you still get or better from which workflow?
The download error is solved in 2.9.1 and the second one, as described earlier, we can't solve anyway if it should reappear again.
For the similar sounding error message while trying to authenticate there's a workaround as described in the [comment above](https://github.com/TumblThreeApp/TumblThree/discussions/383#discussioncomment-4179555).
Please also give us, if possible, the last block of lines in the log file (%LOCALAPPDATA%\TumblThree\TumblThree.log) after you got the error.

**Reply by gr8bigguy** [reply]
20221120 13:50:21.871	Err 	AbstractDownloader:DownloadBinaryFile System.TimeoutException: The operation has timed out.
   at TumblThree.Applications.Extensions.TaskTimeoutExtension.<TimeoutAfter>d__0`1.MoveNext() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Extensions\TaskTimeoutExtension.cs:line 17
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at TumblThree.Applications.Downloader.FileDownloader.<DownloadFileWithResumeAsync>d__12.MoveNext() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Downloader\FileDownloader.cs:line 79
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at TumblThree.Applications.Downloader.AbstractDownloader.<DownloadBinaryFileAsync>d__27.MoveNext() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Downloader\AbstractDownloader.cs:line 80

**Reply by gr8bigguy** [reply]
Error 26 Error while downloading (</Default/Alt>). Please retry shortly.  A low value (1-2) of "Concurrent blogs" is also helpful.
I get this error even when I have 1 or 2 connections.
Error 33:  A connection timeout occurred during downloading

**Reply by thomas694** [reply]
I'm a bit confused...
The extract from the log file shows a timeout error. But your two quoted errors also don't show the initial error.

Out of my own experience, I recommend to use only 1 "concurrent" connection while downloading newTumbl blogs. With that I never got this kind of error the last few weeks. If you just changed to 1 and it still shows errors, you probably have to wait a few minutes or hours.

**Reply by gr8bigguy** [reply]
Tried 1 concurrent connection and still get that error.  I've tried waiting a few days and still got the error.  I will keep trying.

Also, any chance of being able to double-click (or single clicking anywhere) in the Enter URL field to select the entire field to speed up entering the correct newTumbl blog name?

**Reply by thomas694** [reply]
Do you get the error with all blogs or only some?
Does disabling the option "download text posts" in the blog's settings help?
